src/passes/Vacuum.cpp

```diff
@@ -47,11 +49,21 @@ struct Vacuum : public WalkerPass<PostWalker<Vacuum>> {
     walk(func->body);
   }
 
-  // returns nullptr if curr is dead, curr if it must stay as is, or another node if it can be replaced
-  Expression* optimize(Expression* curr, bool resultUsed) {
-    // an unreachable node must not be changed
-    if (curr->type == unreachable) return curr;
+  // Returns nullptr if curr is dead, curr if it must stay as is, or another node if it can be replaced.
+  // Takes into account:
+  //  * The result may be used or unused.
+  //  * The type may or may not matter (a drop can drop anything, for example).
+  Expression* optimize(Expression* curr, bool resultUsed, bool typeMatters) {
+    auto type = curr->type;
+    // An unreachable node must not be changed.
+    if (type == unreachable) return curr;
+    // We iterate on possible replacements. If a replacement changes the type, stop and go back.
+    auto* last = curr;
```

Comment:
Can we call this `previous` or `prev` instead of `last`?

Reply:
Changed.